### DISCUSSION TITLE
Arduino Framework Support

### DISCUSSION THREAD

**Discussion by ZanzyTHEbar** [Fishwaldo/esp_ghota#3] [Ideas]
Fantastic library, using with IDF framework is great. 

Do you mind adding out-of-box arduino framework support? i have some Arduino-based projects i would love to add this to.

**Comment by Fishwaldo** [comment]
Thanks!

I have not used Arduino in years - So I'm pretty rusty on that side. Do you know of any github actions that will compile it to a BIN?

And what specifically would you look for (I imagine a C++ class to wrap it ?) for Ardunio?

**Reply by ZanzyTHEbar** [reply]
Actually, platformio fully supports Arduino - I use platformio for all of my Arduino and non-arduino projects. I would just keep support for Platformio tbh.

Really, just a C++ wrapper class is all it needs plus the `AKConfig` values would need to be build-flags or a header file or a config struct - or even just pass those values into the constructor of the wrapper class.

 If you want to add support for the arduino IDE - that would be tricky since you would have to make a custom action that processed the `ino` into a `cpp` file and then build with the trget architectures compiler. Not worth it imo

**Reply by ZanzyTHEbar** [reply]
In one of my [Arduino projects](https://github.com/lorow/OpenIris/tree/feature/mdns-autodiscovery/ESP), i just used the `ini` to set the `AKConfig` values as build flags - it works - but is a little obtuse.

**Reply by ZanzyTHEbar** [reply]
Here is a [ci/cd github action](https://github.com/lorow/OpenIris/tree/feature/add-ci/.github/workflows) for `platformio` with `arduino framework`

**Comment by Fishwaldo** [comment]
I was just about to fork your repo so I could have a stab at this, and see your already about 8 hours in front of me!
Do you want to open a PR when you have it working and I'll merge it in here? 

(I was going to make some changes around configuring the library, so would be handy if I didn't break your wrapper in the process)

**Reply by ZanzyTHEbar** [reply]
haha, just seeing this now - i actually paused making any kind of wrapper xD after the earlier discussion, as i figured you would make one. The only thing i did, very little, was just implement the raw C code from your lib into my project. which isnt the best solution tbh, it was just to test out the functionality. I can look at making a wrapper - but i have no ETA on that.

**Comment by zekageri** [comment]
That would be awesome!
